javalib/src/main/scala/java/util/package.scala

```diff
@@ -29,7 +31,194 @@ package object util {
   private[util] def defaultOrdering[E]: Ordering[E] = {
     new Ordering[E] {
       def compare(a: E, b: E): Int =
-        a.asInstanceOf[Comparable[E]].compareTo(b)
+        if (a === null) {
+          if (b === null) 0
+          else -1
+        } else {
+          a.asInstanceOf[Comparable[E]].compareTo(b)
+        }
```

Comment:
This is suspicious too. This ordering is not symmetric. `null < foo` returns -1 but `foo < null` throws.

Why is this needed in the first place? JDK collections typically specify that the natural ordering does not support `null` altogether.